src/app/core/services/notifications/notification-wrapper.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@angular/core'
+
+import { DefaultNotificationType } from '../../../../assets/data/defaultConfig'
+import { ConfigKeys } from '../../../shared/enums/config'
+import { NotificationMessagingType } from '../../../shared/models/notification-handler'
+import { RemoteConfigService } from '../config/remote-config.service'
+import { FcmRestNotificationService } from './fcm-rest-notification.service'
+import { FcmXmppNotificationService } from './fcm-xmpp-notification.service'
+import { LocalNotificationService } from './local-notification.service'
+
+@Injectable()
+export class NotificationWrapperService {
+  notificationService:
+    | FcmRestNotificationService
+    | FcmXmppNotificationService
+    | LocalNotificationService
```

Comment:
Instead of ORing these you can just set the type as `NotificationService`, then any sub-class of NotificationService can be used.

Reply:
Made the change.